### PR TITLE
Visitor framework in commands hierarchy.

### PR DESCRIPTION
**Pull request by Backss4**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/IVisitor.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+public interface IVisitor {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/IVisitor.java]: The name should suggest that Visitor concerns DriverCommand hierarchy (e.g., DriverCommandsVisitor)

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/TestVisitor.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.util.Iterator;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class TestVisitor implements IVisitor {
+	private final Logger logger;
+	private int operateToCommandCount = 0;
+	private int setPositionCommandCount = 0;
+	private int compundCommandCount = 0;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/TestVisitor.java]: Provide getters, so visitor can be reused. You can also add:
getCommandsTotalCount
getAtomicCommandsCount

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 3]
```diff
@@ -58,6 +60,10 @@ private static void setupCommandTests(Application application) {
 
 	}
 
+	private static void setupVisitorTests(Application application) {
+		application.addTest("Test Counting Visitor", new SelectTestVisitorOptionListner(logger, new TestVisitor(logger)));
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: Omit "Test" word

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/IElement.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+public interface IElement {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/IElement.java]: I would resign from this interface (add accept directly to DriverCommand). The name and contents is not useful and self-explained.

**Review by kowallus — CHANGES_REQUESTED** [review]
OK, but do code cleaning.

**Comment by Backss4** [comment]
Changes made.

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/CommandCountingVisitor.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+		logger.log(Level.INFO, "Compound Commands count: " + compoundCommandCount);
+	}
+
+	public int getOperateToCommandCount() {
+		return operateToCommandCount;
+	}
+
+	public void setOperateToCommandCount(int operateToCommandCount) {
+		this.operateToCommandCount = operateToCommandCount;
+	}
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/CommandCountingVisitor.java]: I asked for getters only. Setters doesn't make sense here.

**Review by kowallus — CHANGES_REQUESTED** [review]
OK, but please remove setters.

**Pull request merged**